### DISCUSSION TITLE
pytest

### DISCUSSION THREAD

**Discussion by shuther** [BerriAI/litellm#912] [General]
I am getting quite a few errors (or deprecated message) with pytest; is it expected to complete without error (so I can open bug report) or is it still considered work in progress? Maybe the issue are in the tests (i.e. if I don't provide a key for a backend maybe a failure is expected, but shoud it be an internal error?)

**Comment by ghost** [comment]
What errors are you seeing @shuther

**Comment by ghost** [comment]
we use pytest with circle ci to ensure stable builds are being published. Currently we're working through some tests failing. Here's the last working commit that passed pytest - https://github.com/BerriAI/litellm/commit/2686894823cdfd52c82e7c8c74e64f9d7a4ebc56
